pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -1529,6 +1529,15 @@ public static class Accounting {
         "accounting.min.memory.footprint.to.kill.ratio";
     public static final double DEFAULT_MEMORY_FOOTPRINT_TO_KILL_RATIO = 0.025;
 
+    // Per-query memory threshold configurations
+    public static final String CONFIG_OF_ENABLE_PER_QUERY_MEMORY_CHECK = "accounting.per.query.memory.check.enabled";
+    public static final boolean DEFAULT_ENABLE_PER_QUERY_MEMORY_CHECK = false;
+
+    public static final String CONFIG_OF_PER_QUERY_MEMORY_LIMIT_BYTES = "accounting.per.query.memory.limit.bytes";
```

Comment:
Should there be `thread` in the name as single thread's allocation is checked ? Or improve the capability later to consider aggregate ?